lib/livebook/runtime/evaluator.ex

```diff
@@ -675,6 +675,47 @@ defmodule Livebook.Runtime.Evaluator do
   end
 
   defp eval(:erlang, code, binding, env) do
+    is_module = String.starts_with?(code, "-module(")
+
+    case is_module do
+      true -> eval_module(:erlang, code, binding, env)
+      false -> eval_statements(:erlang, code, binding, env)
+    end
+  end
+
+  # Create module - tokens from string
+  # Based on: https://stackoverflow.com/questions/2160660/how-to-compile-erlang-code-loaded-into-a-string
+  # The function will first assume that code starting with -module( is a erlang module definition
+
+  # Step 1: Split the module in \n
+  # Step 2: Scan and convert to froms
+  # Step 3: Extract module name
+  # Step 4: Compile and load
+  defp eval_module(:erlang, code, binding, env) do
+    statements = :binary.split(code, "\n", [:global, :trim_all])
+
+    forms =
+      Enum.map(
+        statements,
+        fn str ->
+          {:ok, result, _} = :erl_scan.string(String.to_charlist(str))
```

Comment:
We need to handle errors accordingly. It will be similar to the `eval_statements` branch, there is probably a lot we can share between the two. :)

Reply:
Thanks for the feedback - will adapt based on the original-code.